.github/workflows/code-freeze.yml

```diff
@@ -32,8 +32,18 @@ jobs:
         echo $BRANCH_NAME
         echo $PR_TITLE
 
-        if [[ "$BRANCH_NAME" != $UNFROZEN_PREFIX* ]] && 
-           [[ "$PR_TITLE" != fix:* && "$PR_TITLE" != *"[critical]"* ]]; then
-          echo "Error: You can only merge from branches that start with '$UNFROZEN_PREFIX', or PRs titled with 'fix: ' and containing '[critical]'."
-          exit 1
+        # if it's not a critical fix
+        if ! [[ "$PR_TITLE" == fix\(critical\):* ]]; then
```

Comment:
Can L36 and L38 be simplified into one conditional? This may make the indentation easier to read

Reply:
No, since L38 makes an if-else around whether there's an unfrozen prefix, but it all falls under L36's condition.

Reply:
```
if ! [[ "$PR_TITLE" == fix\(critical\):* ]] && [[ ! -z $UNFROZEN_PREFIX && "$BRANCH_NAME" != $UNFROZEN_PREFIX* ]]; then
    echo "Error: You can only merge from branches that start with the unfrozen prefix: '$UNFROZEN_PREFIX', or PRs titled with prefix 'fix(critical): '."
    exit 1
fi
```
This condenses the logic a bit to reflect the output message.

Reply:
I think the message might be a bit confusing if unfrozen prefix is not set

Reply:
We could use parameter expansion but that may get tricky if individuals aren't familiar with the syntax